#1108: Add i18n

SteKoe opened the pull request:
Add i18n as requested in #1108

joshiste commented on spring-boot-admin-server-ui/src/main/frontend/index.js:
```diff
@@ -38,6 +41,8 @@ const installables = [
 
+let vueI18Next = new VueI18Next(i18n);
+
 installables.forEach(view => view.install({
```
Would it be somehow possible to pass along the 18n so that translations could be added in custom extensions?

SteKoe replied:
I have added an example to `sample-custom-ui` using i18n :)

joshiste commented on spring-boot-admin-server-ui/src/main/frontend/views/instances/auditevents/auditevents-list.vue:
```diff
@@ -18,11 +18,11 @@
-        <th>Principal</th>
-        <th>Remote address</th>
-        <th>Session Id</th>
+        <th v-html="$t('instances.auditevents.timestamp')" />
```
why is `v-html` needed here?

joshiste commented on spring-boot-admin-server-ui/package.json:
```diff
@@ -14,6 +14,7 @@
     "@fortawesome/free-regular-svg-icons": "^5.8.1",
     "@fortawesome/free-solid-svg-icons": "^5.8.1",
     "@fortawesome/vue-fontawesome": "^0.1.6",
+    "@panter/vue-i18next": "^0.15.0",
```
Are there any reasons why you've chosen this over vue-i18n? Is it because it uses  i18next and doesn't reimplement that stuff?

SteKoe replied:
Nope, just found i18next useful but I will switch over to vue-18n :)

joshiste replied:
Please don’t - I was just curious

joshiste commented on spring-boot-admin-server-ui/src/main/frontend/views/applications/notification-filter-settings.vue:
```diff
@@ -19,7 +19,7 @@
       <div class="field">
         <p class="control has-inline-text">
-          Suppress notifications on <code v-text="object.id || object.name" /> for
+          <span v-html="$t('applications.suppress_notifications_on', {name: object.id || object.name})" />&nbsp;
```
Why it is sometimes `v-t="..."` `v-text="$t(...)"` or `v-html="$t(...)" ... would have expected a consistent way of translating

SteKoe replied:
I have used `v-html` wherever the given text contained HTML. Since it is not safe to use `v-html` I tried to stay away from that wherever possible.

joshiste commented on spring-boot-admin-server-ui/src/main/frontend/i18n/default.de.json:
```diff
@@ -0,0 +1,319 @@
+{
```
Would it be somehow possible to move the translations not in a central file, but ins separate files per view and closer to the views and stitched to gether via `require.conext`? (eg. https://github.com/codecentric/spring-boot-admin/blob/master/spring-boot-admin-server-ui/src/main/frontend/components/index.js#L21)

SteKoe replied:
I will try to do that.

SteKoe replied:
Would it be okay for you to have a file called `i18n.json` aside each and every view? Or would you prefer to have it split up by language as well like `i18n.en.json`?

joshiste replied:
I've no objections on this. I leave this up to you.

SteKoe commented:
I will have to add some more i18n properties as I have missed some vue files.

SteKoe commented:
What did I change:

 * Replace i18next in favor of vue-i18n
 * Added more missing translations to different views (see commit messages for details)
 * Split up i18n file into smaller one-file-per-language-and-view file
 * Added examples for i18n on custom ui 
 * removed usage of `v-t` and replaced it by `v-text`
 * get rid of `{{$t(...)}}` and replaced it by `v-text`

joshiste commented:
@SteKoe I'll revisit this huge PR most probably in 2-3 weeks due to the lack of time.

joshiste commented:
I'd be glad if you could rebase the branch in the meantime...

joshiste commented:
Merged with 0f7c06484add8074c0b0c9c0713d5cd7498826fb and polished with b0a9f284394b9f0e92c8274d366ca674a37493b5
